### ISSUE TITLE
Setting a Host header for a blaze client Request causes an exception to be thrown

### ISSUE THREAD

**Issue by dchenbecker** [http4s/http4s#240]
The issue appears to be faulty logic here:

https://github.com/http4s/http4s/blob/master/blaze-client/src/main/scala/org/http4s/client/blaze/Http1ClientStage.scala#L146

Basically, if Http/1.1 and a "Host" header exists, throw an exception. This seems so obviously wrong to me I feel like I must be missing something. It leads to hilarity like:

```
scala> val reqUri = Uri.fromString("http://foo.com/index.html").fold(_ => throw new Exception("Boo"), identity)
reqUri: org.http4s.Uri = http://foo.com/index.html

scala> val req = Request(uri = reqUri).withHeaders(Header.Host("bar.com"))
warning: there was one deprecation warning; re-run with -deprecation for details
req: org.http4s.Request = Request(GET,http://foo.com/index.html,HTTP/1.1,Headers(Host: bar.com),Halt(End),org.http4s.AttributeMap@1b7e58a2)

scala> defaultClient(req).run
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
java.lang.RuntimeException: Request URI must have a host.
  at scala.sys.package$.error(package.scala:27)
  at org.http4s.client.blaze.Http1ClientStage.org$http4s$client$blaze$Http1ClientStage$$encodeRequestLine(Http1ClientStage.scala:146)
```

If there's agreement that this is wrong, I'm happy to submit a PR to fix it.

**Comment by bryce-anderson** [comment]
This does seem like strange behavior, and I'm sure I'm to blame for it. A PR would be appreciated if you have time, otherwise I can take care of it. I believe we are trying to get a release together very soon to stay in step with scalaz-stream, and it will certainly happen before then.

Out of curiosity, what is the use case for providing your own, assumingly diffierent, host header?

**Comment by dchenbecker** [comment]
https://github.com/http4s/http4s/pull/241 has a fix against master. Would a fix for 0.6.x be useful, too?

**Comment by dchenbecker** [comment]
Use case is connecting to internal hosts behind the load balancer that expect the external host header

**Comment by bryce-anderson** [comment]
Thanks for the PR. Do you need this fixed on the 0.6.x series or can you wait a few days for the 0.7.0 release?

**Comment by dchenbecker** [comment]
I'm fine with th 0.7.0 release, thanks!
